themes/bootstrap3/templates/myresearch/menu.phtml

```diff
@@ -52,6 +52,9 @@
       </a>
     <?php endif; ?>
   <?php endif; ?>
+       <a href="<?=$this->url('overdrive-mycontent')?>"<?=$this->active == 'dgcontent' ? ' class="active"' : ''?>>
```

Comment:
Right now it appears that this menu option is always going to appear -- but presumably we want to hide it when Overdrive is not active.

Reply:
I'm not sure how to handle this one.  I thought maybe there is a way to access the configurations from this template, but I don't see any other template doing that to use as a guide.  Is there a better way to handle this?

Reply:
There is a Config view helper that you can call from a template with `$this->config()->get('config-name')`, so you could do something like `if ($this->config()->get('Overdrive')->Section->Setting ?? false):` to check a config.

We don't use this view helper very much because many functions have their own purpose-built helpers, and it's often cleaner to put a check method on a function-specific helper than to go rooting around in raw configuration files.... but either approach is possible.

Reply:
I'd like to make this a little smarter by allowing the menu to be hidden when the user doesn't have access to Overdrive.  In my "Overdrive Connector" I have a function that will return this.  Is there a way I can access that without going to to much trouble?

Reply:
This is probably an argument for creating an Overdrive view helper, since a view helper is the easiest way to expose aspects of a VuFind service to the templates. Basically you'd just need to create a VuFind\View\Helper\Root\Overdrive class extending the base view helper (and a corresponding OverdriveFactory) and register it in the root theme's theme.config.php. The factory could inject the connector into the helper... and then you could create whatever methods you need. Say, for example, you created `Overdrive::isActive()` -- then in a template you could call `$this->overdrive()->isActive()`.

It's a bit of a headache, but probably the best way. If you like, I could probably stub out the classes for you in this branch fairly quickly and let you fill in / test the details.

Reply:
that would be very much appreciated.

Reply:
See a83d89cb4b57b4aeebd69bf02eb9008f42096304 for an example of how to do this... it might even come close to working, if I guessed correctly what you wanted to do. I'm about out of time for today, but let me know if you have questions/concerns/problems and we can discuss further tomorrow.

I think once this is fully functional, we're going to want to have a bigger conversation about how to name and organize things -- it would probably be better to hide things that reference "Overdrive" behind the more generic "Digital Content" as much as possible to make this more flexible in the long term... but for now I just did it the easy/lazy way so we can get through the "first working prototype" step!

Reply:
Thank you.  this worked just fine.  Are you thinking that there would be a generic DigitalContent view helper?

Reply:
Exactly. I think the ideal situation would be to develop an abstract interface called DigitalContent that could proxy to Overdrive or other services... the challenge, of course, is that until we have more than one use case, it's hard to figure out what the commonalities are. It may make the most sense to implement direct Overdrive access for now and revisit later if a different use case arises... but if we can figure out a way to make it more abstract up front, then it may be less disruptive to increase flexibility in the future. In any case, once there's a fully functional prototype, it may be easier to decide how to proceed (and I realize we're quite close to that point!).